keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/account/FixedAccountCredentialResource.java

```diff
@@ -0,0 +1,234 @@
+import org.keycloak.utils.MediaType;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Copy of {@link org.keycloak.services.resources.account.AccountCredentialResource} with additional resources
+ */
+public class FixedAccountCredentialResource {
+
+    private final KeycloakSession session;
+    private final EventBuilder event;
+    private final UserModel user;
+    private final RealmModel realm;
+    private Auth auth;
+
+    public FixedAccountCredentialResource(KeycloakSession session, EventBuilder event, UserModel user, Auth auth) {
+        this.session = session;
+        this.event = event;
+        this.user = user;
+        this.auth = auth;
+        realm = session.getContext().getRealm();
+    }
+
+    @GET
+    @NoCache
+    @Produces(javax.ws.rs.core.MediaType.APPLICATION_JSON)
+    public List<CredentialRepresentation> credentials(){
+        auth.requireOneOf(AccountRoles.MANAGE_ACCOUNT, AccountRoles.VIEW_PROFILE);
+        List<CredentialModel> models = session.userCredentialManager().getStoredCredentials(realm, user);
+        models.forEach(c -> c.setSecretData(null));
+        return models.stream().map(ModelToRepresentation::toRepresentation).collect(Collectors.toList());
+    }
+
+
+    @GET
+    @Path("registrators")
+    @NoCache
+    @Produces(javax.ws.rs.core.MediaType.APPLICATION_JSON)
+    public List<String> getCredentialRegistrators(){
+        auth.requireOneOf(AccountRoles.MANAGE_ACCOUNT, AccountRoles.VIEW_PROFILE);
+
+        return session.getContext().getRealm().getRequiredActionProviders().stream()
+                .map(RequiredActionProviderModel::getProviderId)
+                .filter(providerId ->  session.getProvider(RequiredActionProvider.class, providerId) instanceof CredentialRegistrator)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Remove a credential for a user
+     *
+     */
+    @Path("{credentialId}")
+    @DELETE
+    @NoCache
+    public void removeCredential(final @PathParam("credentialId") String credentialId) {
+        auth.require(AccountRoles.MANAGE_ACCOUNT);
+        session.userCredentialManager().removeStoredCredential(realm, user, credentialId);
+    }
+
+    /**
+     * Update a credential label for a user
+     */
+    @PUT
+    @Consumes(javax.ws.rs.core.MediaType.TEXT_PLAIN)
+    @Path("{credentialId}/label")
+    public void setLabel(final @PathParam("credentialId") String credentialId, String userLabel) {
+        auth.require(AccountRoles.MANAGE_ACCOUNT);
+        session.userCredentialManager().updateCredentialLabel(realm, user, credentialId, userLabel);
+    }
+
+    /**
+     * Move a credential to a position behind another credential
```

Comment:
Move a credential to the first position ?

Reply:
fixed